auv_teleop/scripts/vehicle_control_tab.py

```diff
@@ -0,0 +1,216 @@
+        control_layout.addWidget(self.right_btn, 1, 2)
+        control_layout.addWidget(self.backward_btn, 2, 1)
+        control_layout.addWidget(self.up_btn, 0, 4)
+        control_layout.addWidget(self.down_btn, 2, 4)
+        control_layout.addWidget(self.yaw_left_btn, 1, 3)
+        control_layout.addWidget(self.yaw_right_btn, 1, 5)
+
+        speed_layout = QHBoxLayout()
+
+        linear_label = QLabel("Linear Speed:")
+        self.linear_speed_spinbox = QDoubleSpinBox()
+        self.linear_speed_spinbox.setRange(0.1, 0.4)
+        self.linear_speed_spinbox.setSingleStep(0.1)
+        self.linear_speed_spinbox.setValue(0.2)
+
+        angular_label = QLabel("Angular Speed:")
+        self.angular_speed_spinbox = QDoubleSpinBox()
+        self.angular_speed_spinbox.setRange(0.1, 0.4)
+        self.angular_speed_spinbox.setSingleStep(0.1)
+        self.angular_speed_spinbox.setValue(0.2)
+
+        speed_layout.addWidget(linear_label)
+        speed_layout.addWidget(self.linear_speed_spinbox)
+        speed_layout.addWidget(angular_label)
+        speed_layout.addWidget(self.angular_speed_spinbox)
+
+        control_vlayout = QVBoxLayout()
+        control_vlayout.addLayout(control_layout)
+        control_vlayout.addSpacing(10)
+        control_vlayout.addLayout(speed_layout)
+        control_group.setLayout(control_vlayout)
+
+        layout.addWidget(teleop_group, 0, 0, 1, 2)
+        layout.addWidget(control_group, 1, 0, 1, 2)
+        self.setLayout(layout)
+
+        self.teleop_start.clicked.connect(self.start_teleop)
+        self.teleop_stop.clicked.connect(self.stop_teleop)
+        self.start_enable.clicked.connect(self.start_enable_publishing)
+        self.stop_enable.clicked.connect(self.stop_enable_publishing)
+
+    def connect_publishing_button(self, button, direction, speed_getter):
+        button.pressed.connect(lambda: self.start_publishing(direction, speed_getter()))
+        button.released.connect(self.stop_publishing)
+
+    def get_linear_spinbox(self):
+        return self.linear_speed_spinbox.value()
+
+    def get_angular_spinbox(self):
+        return self.angular_speed_spinbox.value()
+
+    def start_teleop(self):
+        cmd = ["roslaunch", "auv_teleop", "start_teleop.launch"]
+        if self.xbox_check.isChecked():
+            cmd.append("controller:=xbox")
+        print(f"Executing: {' '.join(cmd)}")
+        self.launch_process = subprocess.Popen(cmd)
+
+    def stop_teleop(self):
+        if self.launch_process is not None:
+            print("Terminating teleop process...")
+            self.launch_process.terminate()
+            try:
+                self.launch_process.wait(timeout=2)
+            except subprocess.TimeoutExpired:
+                print("Process did not terminate, killing it...")
+                self.launch_process.kill()
+            self.launch_process = None
+        else:
+            print("No teleop process to stop.")
+
+    def publish_velocity(self):
```

Comment:
publish_velocity_loop? This function is intended to run continuously
```suggestion
    def publish_velocity_loop(self):
```